src/core/border_router/routing_manager.cpp

```diff
@@ -628,16 +629,23 @@ void RoutingManager::SendRouterAdvertisement(RouterAdvTxMode aRaTxMode)
         SuccessOrExit(error = mRioAdvertiser.AppendRios(raMsg));
     }
 
+    if (Get<InfraIf>().GetLinkLayerAddress(linkAddr) == kErrorNone)
+    {
+        SuccessOrExit(error = raMsg.AppendLinkLayerOption(linkAddr, Option::kSourceLinkLayerAddr));
+    }
+
     if (mExtraRaOptions.GetLength() > 0)
     {
         SuccessOrExit(error = raMsg.AppendBytes(mExtraRaOptions.GetBytes(), mExtraRaOptions.GetLength()));
     }
 
-    VerifyOrExit(raMsg.ContainsAnyOptions());
-
     destAddress.SetToLinkLocalAllNodesMulticast();
     raMsg.GetAsPacket(packet);
 
+    // A valid RA message should contain at lease one option.
+    // Exit when the size of packet is less than the size of header.
+    VerifyOrExit(packet.GetLength() > sizeof(header));
```

Comment:
This check looks the same as the previous removed one, `VerifyOrExit(raMsg.ContainsAnyOptions());` - is there any reason to not use the `ContainsAnyOptions()` here also? It seems clearer that way.